test/module/irohad/simulator/simulator_test.cpp

```diff
@@ -313,14 +330,16 @@ TEST_F(SimulatorTest, RightNumberOfFailedTxs) {
 
   auto proposal_wrapper =
       make_test_subscriber<CallExact>(simulator->on_verified_proposal(), 1);
-  proposal_wrapper.subscribe([&verified_proposal,
-                              &tx_errors](auto verified_proposal_) {
-    // assure that txs in verified proposal do not include failed ones
-    ASSERT_EQ(verified_proposal_->first->height(), verified_proposal->height());
-    ASSERT_EQ(verified_proposal_->first->transactions(),
-              verified_proposal->transactions());
-    ASSERT_TRUE(verified_proposal_->second.size() == tx_errors.size());
-  });
+  proposal_wrapper.subscribe(
+      [&](auto verified_proposal_) {
+        // assure that txs in verified proposal do not include failed ones
+        ASSERT_EQ(verified_proposal_->verified_proposal->height(),
+                  verified_proposal_height);
+        ASSERT_EQ(verified_proposal_->verified_proposal->transactions(),
+                  verified_proposal_transactions);
+        ASSERT_TRUE(verified_proposal_->rejected_transactions.size()
```

Comment:
May be compare hashes as well?

Reply:
Although the test is called `RightNumberOfFailedTxs`, I think you are right and that is a reasonable change. Now the test also checks the hashes, although the checked result is created in this same test.